### ISSUE TITLE
Unable to use jsonb columns with ActiveRecord+Postgres in 3.0.0.beta

### ISSUE THREAD

**Issue by awead** [shrinerb/shrine#407]
### Brief Description

When uploading a file to an ActiveRecord-backed resource using Postgres, there's an error if the column is a `jsonb` type. This problem was not present in 2.19, so I suspect it may be a bug.

### Expected behavior

In 2.19, the file would upload and the resource was saved without issues.

### Actual behavior

When the file is uploaded and the resource saved, it fails on the save with:

```
3: from /Users/agw13/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/shrine-3.0.0.beta/lib/shrine/plugins/column.rb:97:in `load'
2: from /Users/agw13/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/json-2.2.0/lib/json/common.rb:156:in `parse'
1: from /Users/agw13/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/json-2.2.0/lib/json/common.rb:156:in `new'
/Users/agw13/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/json-2.2.0/lib/json/common.rb:156:in `initialize': no implicit conversion of Hash into String (TypeError)
```

### Simplest self-contained example code to demonstrate issue

Unfortunately, you can only reproduce this with a specific set of dependencies. The setup would be something like:

``` ruby
# Add a jsonb column to your table
def change
  add_column :members, :file_data, :jsonb
end

# Define a model class
class Member < ApplicationRecord
  include Shrine::Attachment(:file)
end

# Upload a file to the class
Member.create(file: File.open("test.txt"))
```

The error in question comes from lib/shrine/plugins/column.rb:97 in JsonSerializer#load:

``` ruby
def load(data)
  @json.parse(data)
end
```

`data` at this point is already a Hash instead of a string. You can see the same error in a simple example here when I try to parse a Hash versus a String:

```
> JSON.parse({})
TypeError: no implicit conversion of Hash into String
from /Users/agw13/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/json-2.2.0/lib/json/common.rb:156:in `initialize'
> JSON.parse("{}")
=> {}
```

A simple fix would be to just return data if it's already a hash

``` ruby
def load(data)
  return data if data.is_a?(Hash)

  @json.parse(data)
end
```

I'd be happy to submit a PR if this seems appropriate.

### System configuration

**Ruby version**: 2.6.3

**Shrine version**: 3.0.0.beta

**Comment by janko** [comment]
Thanks for the report. Yes, this part is currently untested, as I didn't want tests to pull in a Postgres dependency, and it got broken in 3.0.0.beta.

The problem isn't in the `column` plugin, but that the `activerecord` plugin doesn't correctly override serialization.

I will push a patch soon, and release 3.0.0.beta2.

**Comment by jrochkind** [comment]
It would be nice to have some automated testing in the area. 

One option would be maybe testing with ActiveRecord's `serialize` to JSON column, on a SQLite (if that's what you're using now for tests) ordinary string/text column. I believe such a test would have failed on this problem. 

Another option would be setting up tests to optionally use postgres, and run that branch of tests on travis (you use travis now?), but you wouldn't have to locally.  The `appraisal` gem could even be used to keep the `pg` gem dependency from being there unless you are testing with that option. I have done this sort of thing before. It would add some complexity around various conditionals around test setup and running. The first option is maybe a better simpler first step. 

@janko let me know if you're interested in either of these approaches, if so I can find some time to PR it.

**Comment by janko** [comment]
@awead Let me know whether the `master` branch works for you now. Then I'll release `3.0.0.beta2` shortly.

@jrochkind I think that adding Postgres would be an overkill, even it runs only on Travis. I've managed to hack JSON column attributes in tests.

**Comment by jrochkind** [comment]
Okay, I'm confused what's going on with that RubySerializer doing eval and to_s -- we definitely don't want to do that in actual shrine code, that's just test code? I think there is an easier and clearer way to do a an AR JSON test using the existing ActiveRecord `serialized` feature, that I think maybe you are trying to reimplement in tests. 

Although I see you are now doing `[:json, :jsonb].include?(column[:type])`, whereas in Shrine 3 I think it did `value.kind_of?(Hash)` instead? You think the new way is superior? I kind of like the reliability of kind_of Hash. 

Still willing to do a PR with tests that use AR `serialized` feature if you are interested!

The important thing in live code is that shrine does _not_ parse/serialize the columns that AR is providing already as ruby hashes, but just takes them as they are, rather than dumps them and re-serializes them or something.

**Comment by awead** [comment]
@janko `master` is working for me. Thanks for taking care of this!

**Comment by janko** [comment]
> Okay, I'm confused what's going on with that RubySerializer doing eval and to_s

@jrochkind `RubySerializer` is just something we use for testing column serializer, it's not related to testing JSON columns. I see that these changes accidentally got merged together with the changes related to JSON column support, my mistake.

> Although I see you are now doing [:json, :jsonb].include?(column[:type]), whereas in Shrine 3 I think it did value.kind_of?(Hash) instead? You think the new way is superior? I kind of like the reliability of kind_of Hash.

I'm not sure what you're referring to. We always checked for JSON columns like this.

**Comment by janko** [comment]
@awead I've just released 3.0.0.beta2 which includes this fix.
